### PR TITLE
bugfix | signout signin redirect issue

### PR DESCRIPTION
**Pull request by omarCreativeDev**
- check if logged in user is the same as previously stored establishmentId
- then conditionally set redirect if logged in user hasn't changed

### CODE REVIEW

**Comment by omarCreativeDev** [comment]
[(2) Defect - Sign out/in issue - Front End](https://trello.com/c/sja7Acds/38-2-defect-sign-out-in-issue-front-end)

File: src/app/core/services/establishment.service.ts [hunk 1]
```diff
@@ -39,11 +39,27 @@ interface EmployerTypeRequest {
...
   public establishment$: Observable<Establishment> = this._establishment$.asObservable();
+  public previousEstablishmentId: number;
+  public isSameLoggedInUser: boolean;
 
   constructor(private http: HttpClient) {}
 
   private _establishmentId: number = null;
 
+  public checkIfSameLoggedInUser(establishmentId: number): void {
+    if (!this.previousEstablishmentId) {
```
Comment (ChazUK) [inline src/app/core/services/establishment.service.ts]: What happens if I login to 3 different accounts? Looks like this will only run for the first time.

Comment (omarCreativeDev) [reply src/app/core/services/establishment.service.ts]: good spot @ChazUK, i have tweaked the pr and tested thoroughly and the fix seems to be working fine

File: src/app/core/services/establishment.service.ts [hunk 2]
```diff
@@ -39,11 +39,23 @@ interface EmployerTypeRequest {
...
 
   constructor(private http: HttpClient) {}
 
   private _establishmentId: number = null;
 
+  public checkIfSameLoggedInUser(establishmentId: number): void {
+    if (!this.previousEstablishmentId || this.previousEstablishmentId !== establishmentId) {
+      this.previousEstablishmentId = establishmentId;
+      this.isSameLoggedInUser = false;
+      return;
```
Comment (ChazUK) [inline src/app/core/services/establishment.service.ts]: I don't think there's a need for this return anymore

**Pull request merged**